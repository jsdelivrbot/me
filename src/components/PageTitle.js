import React from 'react';
import PropTypes from 'prop-types';
import { Col, Container, Row } from 'reactstrap';

import '../styles/components/PageTitle.css';

const PageTitle = props => {
  const {
    title,
    subtitle
  } = props;
  return (
    <div className="PageTitle">
      <Container>
        <Row>
          <Col className="text-md-left" xs="12" md="10">
            <h3>
              { title }
              {subtitle &&
                <small className="text-muted">{ subtitle }</small>
              }
            </h3>
          </Col>
          <Col xs="12" md="2">
            { props.right }
          </Col>
        </Row>
      </Container>
    </div>
  );
};

PageTitle.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
};

export default PageTitle;