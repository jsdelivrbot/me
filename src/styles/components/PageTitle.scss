.PageTitle {
  padding-top: 16px;
}