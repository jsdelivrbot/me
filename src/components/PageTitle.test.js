import React from 'react';
import renderer from 'react-test-renderer';

import PageTitle from './PageTitle';

describe('PageTitle', () => {
  const makeComponent = customProps => {
    const defaultProps = {
      title: 'Foo',
      subtitle: 'Bar'
    };
    const props = { ...defaultProps, ...customProps };
    return renderer.create(
      <PageTitle { ...props } />,
    );
  };

  test('_SNAPSHOT_', () => {
    const component = makeComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render without crashing', () => {
    const component = makeComponent();
    expect(component).toBe.ok;
  });
});