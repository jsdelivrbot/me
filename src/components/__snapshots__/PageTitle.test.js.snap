// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle _SNAPSHOT_ 1`] = `
<div
  className="PageTitle"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="text-md-left col"
      >
        <h3>
          Foo
          <small
            className="text-muted"
          >
            Bar
          </small>
        </h3>
      </div>
    </div>
  </div>
</div>
`;
